File: IKeep.Lib/Models/Relations/GenericElementGenericChore.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace IKeep.Lib.Models
{
    public class GenericElementGenericChore : Entity
    {
        public Guid GenericElementId { get; set; }
        [JsonIgnore]
        public virtual GenericElement GenericElement { get; set; }

        public string GenericElementName
        {
            get
            {
                return GenericElement == null ? "" : GenericElement.Name;
            }
        }
        public Guid GenericChoreId { get; set; }
        [JsonIgnore]
        public virtual GenericChore GenericChore { get; set; }
        public string GenericTaskDescription
        {
            get
            {
                return GenericChore == null ? "" : GenericChore.Description;
            }
        }
        public Status Status { get; set; }
    }

    public enum Status
    {
        Inactive,
        Active
    }
}

File: IKeep.Web/ClientApp/src/models/GenericElementGenericChore.js
import Entity from './Entity'

class GenericElementGenericChore extends Entity
{
    constructor(json)
    {
        super(json);
        if(json)
        {
            this.GenericElementId = json.genericElementId;
            this.GenericChoreId = json.genericChoreId;
            this.Status = json.status;
            this.GenericElementName = json.generiElementName;
            this.GenericChoreDescription = json.genericChoreDescription;
        }
        else
        {
            this.GenericElementId = "";
            this.GenericChoreId = "";
            this.Status = 1;
            this.GenericElementName = "";
            this.GenericChoreDescription = "";
        }
        
    }
}

export default GenericElementGenericChore;